nginx-sys/build.rs

```diff
@@ -196,34 +201,66 @@ build process:
+fn zlib_archive_url(version: &String) -> String {
     format!("{ZLIB_DOWNLOAD_URL_PREFIX}/v{version}/zlib-{version}.tar.gz")
 }
 
-fn pcre2_archive_url() -> String {
-    let version = env::var("PCRE2_VERSION").unwrap_or_else(|_| PCRE2_DEFAULT_VERSION.to_string());
-    format!("{PCRE2_DOWNLOAD_URL_PREFIX}/pcre2-{version}/pcre2-{version}.tar.gz")
+fn pcre_archive_url(version: &String) -> String {
+    // We can distinguish pcre1/pcre2 by checking whether the second character is '.', because the final version of pcre1 is 8.45 and the first one of pcre2 is 10.00.
+    if version.chars().nth(1).is_some_and(|c| c == '.') {
+        format!("{PCRE1_DOWNLOAD_URL_PREFIX}/{version}/pcre-{version}.tar.gz")
+    } else {
+        format!("{PCRE2_DOWNLOAD_URL_PREFIX}/pcre2-{version}/pcre2-{version}.tar.gz")
+    }
 }
 
-fn openssl_archive_url() -> String {
-    let version = env::var("OPENSSL_VERSION").unwrap_or_else(|_| OPENSSL_DEFAULT_VERSION.to_string());
-    format!("{OPENSSL_DOWNLOAD_URL_PREFIX}/openssl-{version}/openssl-{version}.tar.gz")
+fn openssl_archive_url(version: &String) -> String {
+    if version.starts_with("1.1.1") {
+        let version_hyphened = version.replace(".", "_");
+        format!("{OPENSSL_DOWNLOAD_URL_PREFIX}/OpenSSL_{version_hyphened}/openssl-{version}.tar.gz")
+    } else {
+        format!("{OPENSSL_DOWNLOAD_URL_PREFIX}/openssl-{version}/openssl-{version}.tar.gz")
+    }
 }
 
-fn nginx_archive_url() -> String {
-    let version = env::var("NGX_VERSION").unwrap_or_else(|_| NGX_DEFAULT_VERSION.to_string());
+fn nginx_archive_url(version: &String) -> String {
     format!("{NGX_DOWNLOAD_URL_PREFIX}/nginx-{version}.tar.gz")
 }
 
 /// Returns a list of tuples containing the URL to a tarball archive and the GPG signature used
 /// to validate the integrity of the tarball.
 fn all_archives() -> Vec<(String, String)> {
+    let ngx_version = env::var("NGX_VERSION").unwrap_or_else(|_| NGX_DEFAULT_VERSION.into());
+    let zlib_version = env::var("ZLIB_VERSION").unwrap_or_else(|_| ZLIB_DEFAULT_VERSION.into());
+    // keep env name `PCRE2_VERSION` for compat
+    // Nginx had started supporting pcre2 and openssl3 since 1.22.0
+    let ngx_version_vec: Vec<i16> = ngx_version.split('.').map(|s| s.parse().unwrap_or(-1)).collect();
+    let is_after_1_22 = (ngx_version_vec.len() >= 2)
+        && (ngx_version_vec[0] > 1 || (ngx_version_vec[0] == 1 && ngx_version_vec[1] >= 22));
+    let pcre_version = env::var("PCRE2_VERSION").unwrap_or_else(|_| {
+        if is_after_1_22 {
+            PCRE2_DEFAULT_VERSION
+        } else {
+            PCRE1_DEFAULT_VERSION
+        }
+        .into()
+    });
+    let openssl_version = env::var("OPENSSL_VERSION").unwrap_or_else(|_| {
+        if is_after_1_22 {
+            OPENSSL3_DEFAULT_VERSION
+        } else {
+            OPENSSL1_DEFAULT_VERSION
+        }
+        .into()
+    });
+    fn easy_make_url_pair(tar_url: String, pgp_ext: &str) -> (String, String) {
```

Comment:
What does "easy" and "make" mean in this context? Wouldn't just `url_pair` be sufficient? Perhaps this is a style nit, but I'm just a little confused if "easy make" is referring to a specific tool or pattern.

Reply:
I just added `make` to turn the name into a verb phrase, and `easy` to mean easy-going, because the URL of the tarball isn't always a substring of the URL of the PGP key. Nevertheless, this fn is a local one, not polluting the global namespace, so `url_pair()` is enough, I agree.